### ISSUE TITLE
[Bug] No way to set colors for Time Picker, Date Picker, Picker, DisplayAlert popups

### ISSUE THREAD

**Issue by mfeingol** [xamarin/Xamarin.Forms#11616]
### Description

The popups used by Time Picker, Date Picker, Picker, and DisplayAlert cannot be themed using any Xamarin.Forms constructs.

### Steps to Reproduce

N/A

### Expected Behavior

It should be possible to use styles to customize the colors used in these popup dialogs.

### Actual Behavior

There is no way to do this in Xamarin Forms.

### Basic Information

- Version with issue: All
- Last known good version: None
- IDE: VS 2019
- Platform Target Frameworks: 
  - Android: 10
- Android Support Library Version: Latest

### Screenshots

N/A

### Reproduction Link

N/A

### Workaround

It appears to be possible to style these popups from styles.xml, but that's a poor workaround because it can't be done programmatically.

**Comment by PureWeen** [comment]
related
https://github.com/xamarin/Xamarin.Forms/issues/11652

**Comment by PrimoDev23** [comment]
@mfeingol are you talking about android specific or cross platform? On android it is possible as long as u don't use Visual Material and stay with appcompat
It is possible with material too but u can't switch themes dynamically anymore then

**Comment by mfeingol** [comment]
The only mechanism I'm aware of is what I mentioned under `Workaround` above. Is there a different mechanism?

**Comment by PrimoDev23** [comment]
> The only mechanism I'm aware of is what I mentioned under `Workaround` above. Is there a different mechanism?

Well, I somehow missed the workaround section. It is possible with styles.xml and works fine actually (and not that bad workaround because we are on native ways, so this should be harder but cleaner at all)
I tried using a custom renderer which wasn't working that good. I currently go with styles.xml theme and switching themes dynamically by using a dependencyservice

**Comment by mfeingol** [comment]
Yeah, I'd be happy to write a renderer if needed, but I haven't found the native Android APIs to do this. I was hoping I was missing something.

I think the ask here is for dynamic on-the-fly theme switching, ideally using Xamarin Forms APIs.

**Comment by PrimoDev23** [comment]
> Yeah, I'd be happy to write a renderer if needed, but I haven't found the native Android APIs to do this. I was hoping I was missing something.
> 
> I think the ask here is for dynamic on-the-fly theme switching, ideally using Xamarin Forms APIs.

Renderer for Picker at example is either PickerRenderer or MaterialPickerRenderer. U can open a number picker Dialog or whatever on click but in my Tests it looked totally different from what XF uses by default. I think the best way would be to create a custom control or wait until its added.

Under android u can switch themes/styles at runtime with this line

`MainActivity.Instance.SetTheme(theme == Theme.Light ? Resource.Style.MainTheme_Base : Resource.Style.MainTheme_BaseDark);`

And using dynamicresources ofc
That's also working for dialogs as long as u don't use Visual Material on Dialog components, I can send some examples if u want

**Comment by mfeingol** [comment]
@PrimoDev23: thanks, calling `SetTheme` is a fine platform-specific workaround for now. Now sure how I'd missed that earlier.
